futurex_openedx_extensions/helpers/export_data.py

```diff
@@ -0,0 +1,179 @@
+        response_data = {'download_url': '', 'error': ''}
+        filtered_query_params = self.get_filtered_query_params()
+        view_url = self.get_view_request_url(filtered_query_params)
+        view_data = {
+            'user_id': self.request.user.id,
+            'view_class_name': self.__class__.__name__,
+            'query_params': filtered_query_params,
+            'kwargs': self.kwargs,
+        }
+
+        error = None
+        try:
+            csv_file, error = export_data_to_csv(
+                view_url, view_data, self.request.fx_permission_info, self.filename
+            )
+            response_data['download_url'] = self.build_download_url(csv_file)
+        except Exception as exc:  # pylint: disable=broad-exception-caught
+            response_data['error'] = f'Failed to write CSV file due to internal error: {str(exc)}'
+
+        if error:
+            response_data['error'] = error
+
+        return response_data
+
+    def list(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        """Override the list method to generate CSV and return JSON response with CSV URL"""
+        if self.request.query_params.get('download') == 'csv':
+            response = self.generate_csv_url_response()
+            return Response(response, status=http_status.HTTP_200_OK)
+
+        return super().list(request, args, kwargs)
+
+
+def _get_mocked_request(url: str, user: Any, fx_info: dict, query_params: dict) -> Request:
+    """Create mocked request"""
+    factory = APIRequestFactory()
+    mocked_request = factory.get(url)
+    mocked_request.user = user
+    mocked_request.fx_permission_info = fx_info
+    mocked_request.query_params = query_params
+    return mocked_request
+
+
+def _get_user(user_id: int) -> Any:
+    """Get User from user_id"""
+    if user_id:
+        try:
+            return User.objects.get(id=user_id)
+        except User.DoesNotExist:
+            # Error handling need to be done
+            pass
+    return None
+
+
+def _get_view_class_instance(view_name: str) -> Any:
+    """Create view class instance from given iew name"""
+    view_instance = None
+    dasboard_views_module = importlib.import_module('futurex_openedx_extensions.dashboard.views')
+    view_class = getattr(dasboard_views_module, view_name, None)
+    if view_class:
+        view_instance = view_class.as_view()
+    return view_instance
+
+
+def export_data_to_csv(url: str, view_data: dict, fx_permission_info: dict, filename: str) -> tuple:
+    """
+    Mock view with given view params and write JSON response to CSV
+
+    :param url: view url
+    :param view_data: required data for mocking
+    :param fx_permission_info: contains role and permission info
+    :param fx_permission_info: filename for generated CSV
+
+    :return: generated filename
+    """
+    user = _get_user(view_data.get('user_id', None))
+    if not user:
+        return ('', f'Unable to get user for user_id: {view_data.get("user_id")}')
```

Comment:
Many places in the code handle errors with silent returns. Please consider explicit `raise` instead. Having an exception in the logs is better than debugging and tracing to find the reason for failure. You might take advantage of `FXCodedException` by using existing error code or adding new ones

```suggestion
        raise FXCodedException(
            FXExceptionCodes.USER_NOT_FOUND,
            'CSV Export: invalid user_id!'
        )
```

on the other hand, this particular exception should be inside `_get_user`